geckomat/change_model/getComplexData.m

```diff
@@ -0,0 +1,139 @@
+    if strcmp(organism,'Saccharomyces cerevisiae')
+        organism = 'Saccharomyces cerevisiae (strain ATCC 204508 / S288c)';
+    elseif strcmp(organism,'Schizosaccharomyces pombe')
+        organism = 'Schizosaccharomyces pombe (strain 972 / ATCC 24843)';
+    elseif strcmp(organism,'Escherichia coli')
+        organism = 'Escherichia coli (strain K12)';
+    end 
+end
+
+try
+    url1 = 'https://www.ebi.ac.uk/intact/complex-ws/search/*';
+    if ~isempty(organism)
+        url1 = [url1 '?facets=species_f&filters=species_f:("' organism '")'];
+    end
+
+    data = webread(url1);
+
+catch ME
+    if (strcmp(ME.identifier,'MATLAB:webservices:HTTP404StatusCodeError'))
+        disp('Cannot connect to the complex portal, perhaps the server is not responding');
+    end
+end
+
+complexData = cell(data.size,6);
+for i = 1:data.size
+
+    url2 = 'https://www.ebi.ac.uk/intact/complex-ws/complex/';
+    complexID = data.elements(i,1).complexAC;
+
+    disp(['Retrieving information for ' complexID '. Complex ' int2str(i) ' of ' int2str(data.size)]);
+
+    try
+        temp = webread([url2 complexID]);
+    catch ME
+        if (strcmp(ME.identifier,'MATLAB:webservices:HTTP404StatusCodeError'))
+            warning(['Cannot retrieve the information for ' complexID]);
+        end
+        temp = [];
+    end
+
+    if ~isempty(temp)
+        complexData(i,1) = {temp.complexAc};
+        complexData(i,2) = {temp.name};
+        complexData(i,3) = {temp.species};
+
+        idxIntType = find(strcmpi({temp.participants.interactorType}, 'protein'));
+        
+        % Some complex reported are 'stable complex', then, save the id
+        % complex and but set the genes and protein to a empty string.
+        if numel(idxIntType) > 0 
+            complexData(i,4) = {{temp.participants(idxIntType).name}};
+            complexData(i,5) = {{temp.participants(idxIntType).identifier}};
+        else
+            complexData(i,4) = {' '};
+            complexData(i,5) = {' '};
+        end
+
+        % Portal complex has two stochiometry values, a minimum and
+        % maximum value. Only minimum will be store. In some cases,
+        % some complex does not have stochiometry coefficient, then, it
+        % will be fill with zeros
+        if ~cellfun('isempty',{temp.participants.stochiometry})
+            % For some reason, if there is only one protein in the complex
+            % split function does nor return a cell nx2, instead is 2x1,
+            % then assign an incorrect stochiometry 
+            if numel(idxIntType) > 1
+                stochiometry = split({temp.participants(idxIntType).stochiometry}.', ',');
+            else
+                stochiometry = split({temp.participants(idxIntType).stochiometry}.', ',').';
+            end 
+            values = str2double(erase(stochiometry(:,1),"minValue: ")).';
+            complexData(i,6) = {values};
+        else
+            complexData(i,6) = {repelem(0,numel(complexData{i,4}))};
```

Comment:
```suggestion
            complexData(i,6) = {repelem(1,numel(complexData{i,4}))};
```
If no specific stochiometry is given, should it not be filled with 1s instead of 0s? For [CPX-3248](https://www.ebi.ac.uk/complexportal/complex/CPX-3248) no stochiometry is given, in contrast to e.g. [CPX-3238](https://www.ebi.ac.uk/complexportal/complex/CPX-3238). By setting 0s and running `applyComplexData` afterwards results in subunits no longer being required for those reactions. By setting 1s we just keep the status quo: we do not know the exact stochiometry so we assume 1:1.

Reply:
Yes, I think about that, I keep to zero in order to notify the user about it. But maybe can be another extra column mentioning that status quo: so user can take this into account for potential manual curation

Reply:
Ah, I hadn't noticed you actually deal with this when complex data is applied to the model:

https://github.com/SysBioChalmers/GECKO/blob/90f1f9171b10930374a1b30f3b278cc9ba7a0036/geckomat/change_model/applyComplexData.m#L97-L101

But indeed, let's include a `stochiometryDefined` column (or similar) in the `foundComplex`, `proposedComplex` and `complexInfo` data structures, and in the `complex_data.json` file.